datafusion/sql/src/select.rs

```diff
@@ -477,6 +477,15 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
 
                 let filter_expr =
                     self.sql_to_expr(predicate_expr, plan.schema(), planner_context)?;
+
+                // Check for aggregation functions
+                let aggregate_exprs = find_aggregate_exprs(&[filter_expr.clone()]);
```

Comment:
```suggestion
                let aggregate_exprs =
                    find_aggregate_exprs(std::slice::from_ref(&filter_expr));
```
To avoid cloning